GitCommands/RevisionGraph.cs

```diff
@@ -186,12 +186,15 @@ private void ProccessGitLog(CancellationToken taskState)
                 (BranchFilter.IndexOfAny(ShellGlobCharacters) >= 0))
                 branchFilter = "--branches=" + BranchFilter;
 
+            string revisionsToPathsSeparator = Filter.Contains(" -- ") ? string.Empty : " --"; // When there is a path filter it already contains the separator
```

Comment:
I think that would require to split filter into some structure more complex than a string as a filter can contain params such as `--max-count="100000"` that come before path but also optionally path/filename filters which do require ` -- ` separation.